core/cmd/dkg_setup.go

```diff
@@ -0,0 +1,245 @@
+
+		select {
+		case <-cleanExit:
+			return
+		case <-time.After(cli.Config.ShutdownGracePeriod()):
+		}
+
+		lggr.Criticalf("Shutdown grace period of %v exceeded, closing DB and exiting...", cli.Config.ShutdownGracePeriod())
+		// LockedDB.Close() will release DB locks and close DB connection
+		// Executing this explicitly because defers are not executed in case of os.Exit()
+		if err = ldb.Close(); err != nil {
+			lggr.Criticalf("Failed to close LockedDB: %v", err)
+		}
+		if err = cli.CloseLogger(); err != nil {
+			log.Printf("Failed to close Logger: %v", err)
+		}
+
+		os.Exit(-1)
+	})
+
+	if err = ldb.Open(rootCtx); err != nil {
+		return nil, cli.errorOut(errors.Wrap(err, "opening db"))
+	}
+	defer lggr.ErrorIfClosing(ldb, "db")
+
+	app, err := cli.AppFactory.NewApplication(cli.Config, ldb.DB())
+	if err != nil {
+		return nil, cli.errorOut(errors.Wrap(err, "fatal error instantiating application"))
+	}
+
+	keyStore := app.GetKeyStore()
+	err = cli.KeyStoreAuthenticator.authenticate(c, keyStore)
+	if err != nil {
+		return nil, errors.Wrap(err, "error authenticating keystore")
+	}
+
+	evmChainSet := app.GetChains().EVM
+	if cli.Config.EVMEnabled() {
+		if err != nil {
+			return nil, errors.Wrap(err, "error migrating keystore")
+		}
+
+		for _, ch := range evmChainSet.Chains() {
+			err2 := app.GetKeyStore().Eth().EnsureKeys(ch.ID())
+			if err2 != nil {
+				return nil, errors.Wrap(err2, "failed to ensure keystore keys")
+			}
+		}
+	}
+
+	err2 := app.GetKeyStore().OCR2().EnsureKeys()
+	if err2 != nil {
+		return nil, errors.Wrap(err2, "failed to ensure ocr key")
+	}
+
+	err2 = app.GetKeyStore().DKGSign().EnsureKey()
+	if err2 != nil {
+		return nil, errors.Wrap(err2, "failed to ensure ocr key")
+	}
+
+	err2 = app.GetKeyStore().DKGEncrypt().EnsureKey()
+	if err2 != nil {
+		return nil, errors.Wrap(err2, "failed to ensure ocr key")
+	}
+
+	err2 = app.GetKeyStore().P2P().EnsureKey()
+	if err2 != nil {
+		return nil, errors.Wrap(err2, "failed to ensure p2p key")
+	}
+
+	dkgEncrypt, _ := app.GetKeyStore().DKGEncrypt().GetAll()
+	dkgSign, _ := app.GetKeyStore().DKGSign().GetAll()
+	p2p, _ := app.GetKeyStore().P2P().GetAll()
+	ocr2List, _ := app.GetKeyStore().OCR2().GetAll()
+	ethKeys, _ := app.GetKeyStore().Eth().GetAll()
+
+	peerID := p2p[0].PeerID().Raw()
+	if c.Bool("isBootstrapper") == false {
+		peerID = c.String("bootstrapperPeerID")
+	}
+	ocr2 := ocr2List[0]
+	offChainPublicKey := ocr2.OffchainPublicKey()
+	configPublicKey := ocr2.ConfigEncryptionPublicKey()
+	keyID := c.String("keyID")
+	transmitter := ethKeys[0].Address.String()
+	dkgEncryptKey := dkgEncrypt[0].PublicKeyString()
+	dkgSignKey := dkgSign[0].PublicKeyString()
+
+	if c.Bool("isBootstrapper") {
+		sp := fmt.Sprintf(`
+		type = "bootstrap"
+		schemaVersion = 1
+		name = ""
+		externalJobID = "4a48cc1b-9091-465b-9f78-2539341009d1"
+		id = "3"
+		contractID = "%s"
+		relay = "evm"
+		monitoringEndpoint = "chain.link:4321"
+		
+		[relayConfig]
+		chainID = 4
```

Comment:
nit-picky comment: I'd pull these out as constants and name them `bootstrapperJobSpecTemplate` and `dkgJobSpecTemplate`

Reply:
yuuuup